### ISSUE TITLE
Code Quality: Cancel loading extended properties when items are scrolled out of viewport

### ISSUE THREAD

**Issue by yaira2** [files-community/Files#14675]
### Description

We only load extended properties for up to 50 items at a time, this is good because it helps avoid overloading the processor and crashing the app. However, this also means that if you scroll too fast, you may experience some delay in loading the extended properties for the visible items.
We might be able to optimize this by canceling the loading of extended properties when items are scrolled out of the viewport. It's also worth checking if it helps to add a slight delay (eg. 100ms) before loading extended properties so that the processes isn't started unless the user is scrolling slowly.

### Concerned code

ItemViewModel

### Gains

- Reduces the load on the processor when scrolling fast
- Loads extended properties for items in the viewport quicker

**Comment by workbysaran** [comment]
@yair100 

Agree! When an item enters the viewport, we can add a 100ms delay before starting any real work and this naturally handles fast scrolling since items that leave the viewport within that window get cancelled immediately via a per-item CancellationTokenSource without ever touching the semaphore.

**Comment by yair100** [comment]
I don't know if this will really work, but let's give it a try.
